fs/CharDevice.h

```
@@ -0,0 +1,26 @@
+#include <stdint.h>
+#include <stddef.h>
+
+#define MAX_CHAR_DEVICES 32
+
+struct CharDevice;
+
+typedef int (*CharReadFunc)(struct CharDevice* dev, void* buffer, uint32_t size);
+typedef int (*CharWriteFunc)(struct CharDevice* dev, const void* buffer, uint32_t size);
+
+typedef struct CharDevice {
+    char name[32];
```

Clang (14.0.6)

[warning] 32 is a magic number; consider replacing it with a named constant

(cppcoreguidelines-avoid-magic-numbers,readability-magic-numbers)